plugins/org.jkiss.dbeaver.ui.editors.base/src/org/jkiss/dbeaver/ui/editors/object/struct/EditForeignKeyPage.java

```diff
@@ -681,6 +672,42 @@ protected void setSelectedNode(DBNDatabaseNode node) {
         }
     }
 
+    @Nullable
+    private DBPProject getOwnerProject() {
+        DBPProject ownerProject = navigatorModel.getRoot().getOwnerProject();
+        if (ownerProject != null) {
+            return ownerProject;
+        }
+        if (ownerContainerNode != null) {
+            return ownerContainerNode.getOwnerProject();
+        }
+        if (ownerTableNode != null) {
+            return ownerTableNode.getOwnerProject();
+        }
+        if (foreignKey != null) {
+            DBNDatabaseNode node = getNodeByContainerObject(foreignKey.getParentObject());
+            if (node != null) {
+                node.getOwnerProject();
+            }
+        }
+        return null;
+    }
+
+    @Nullable
```

Comment:
Looks like over-engineering.

Reply:
Here I just extracted code to the method to reuse it. I don't see why it could be over-engineering.